### PR TITLE
Better YAML format

### PR DESCRIPTION
**Pull request by cdupuis**
This moves the YAML format to a more consise and consistent syntax. See @atomist/sdm#791

### CODE REVIEW

File: lib/machine/yaml/configureYaml.ts [hunk 1]
```diff
@@ -241,25 +239,8 @@ async function createGoalData<G extends DeliveryGoals>(patterns: string | string
...
-                            options.tests || {},
-                            testMakers);
-                        const dependsOn = value.dependsOn || value.depends_on;
-
-                        goalData[k] = {
-                            test: toArray(test).length > 0 ? test : undefined,
-                            dependsOn,
-                            goals,
-                        };
+                    if (k === "rules") {
```
Comment (ddgenome) [inline lib/machine/yaml/configureYaml.ts]: That's much nicer.

File: lib/machine/yaml/configureYaml.ts [hunk 2]
```diff
@@ -271,7 +252,8 @@ async function createGoalData<G extends DeliveryGoals>(patterns: string | string
 
 async function requireExtensions<EXT>(cwd: string,
                                       pattern: string[],
-                                      cb: (v: EXT, k: string, e: Record<string, EXT>) => void = () => { },
+                                      cb: (v: EXT, k: string, e: Record<string, EXT>) => void = () => {
+                                      },
```
Comment (ddgenome) [inline lib/machine/yaml/configureYaml.ts]: ?

File: lib/machine/yaml/mapGoals.ts [hunk 3]
```diff
@@ -81,83 +78,89 @@ const MapContainer: MapGoal = async (goals: any,
...
-        goal = container(
-            name,
+        const g = container(
+            goals.name,
             {
                 callback: containerCallback(),
                 containers,
-                volumes: camelCase(toArray(goals.volumes)),
-                input: toArray(goals.input),
-                output: mapOutput(goals),
```
Comment (ddgenome) [inline lib/machine/yaml/mapGoals.ts]: So the handling of input/output on the fulfillment is now all that is needed?

File: lib/machine/yaml/configureYaml.ts (hunk 2 above)
Comment (cdupuis) [reply lib/machine/yaml/configureYaml.ts]: That’s just some wild auto format.

File: lib/machine/yaml/mapGoals.ts (hunk 3 above)
Comment (cdupuis) [reply lib/machine/yaml/mapGoals.ts]: The inputs and outputs are registered for any yaml goal now in addCaching. Now this is consistent with all goals coming from yaml.

**Pull request merged**